devtools/server/actors/replay/connection.js

```diff
@@ -185,7 +185,7 @@ class Recording extends EventEmitter {
 
   async _onFinished(data) {
     try {
-      this.emit("finished");
+      this.emit("finished", data);
```

Comment:
Do we know how the front end behaves if we emit `finished` here and then run into an error and emit `unusable` via the `catch` block below?

Reply:
If I force an error before calling `Internal.setRecordingMetadata`, the recorded window shows the following and the playback tab skips the upload screen and seems to hang. I'm not sure if this is a representative scenario but is probably something to follow up on if nothing else.

![image](https://user-images.githubusercontent.com/788456/118536521-b0430f80-b700-11eb-8302-d48493d00877.png)

Reply:
good call. We probably want to replace this with a nicer upload screen error page.

https://github.com/RecordReplay/gecko-dev/issues/368